controllers/volsync/vshandler.go

```diff
@@ -1531,36 +1738,60 @@ func (v *VSHandler) IsRDDataProtected(pvcName string) (bool, error) {
 }
 
 func (v *VSHandler) SelectDestCopyMethod(rdSpec ramendrv1alpha1.VolSyncReplicationDestinationSpec, log logr.Logger,
-) (volsyncv1alpha1.CopyMethodType, *string, error) {
-	if !v.IsCopyMethodDirect() {
-		v.log.Info("Using default copyMethod of Snapshot")
+) (volsyncv1alpha1.CopyMethodType, *string, bool, error) {
+	const requeue = true
 
-		return volsyncv1alpha1.CopyMethodSnapshot, nil, nil // use default copyMethod
-	}
+	if v.IsCopyMethodSnapshot() {
+		v.log.Info("Using copyMethod of Snapshot")
 
-	// IF using CopyMethodDirect, then ensure that the PVC exists, otherwise, create it.
-	err := v.EnsurePVCforDirectCopy(v.ctx, log, rdSpec)
-	if err != nil {
-		return "", nil, err
-	}
+		return volsyncv1alpha1.CopyMethodSnapshot, nil, !requeue, nil // use default copyMethod
+	} else if v.IsCopyMethodDirect() {
+		v.log.Info("Using copyMethod of Direct")
+		// IF using CopyMethodDirect, then ensure that the PVC exists, otherwise, create it.
+		err := v.EnsurePVCforDirectCopy(v.ctx, log, rdSpec)
+		if err != nil {
+			return "", nil, requeue, err
+		}
 
-	// PVC must not be in-use before creating the RD
-	inUse, err := v.isPVCInUseByNonRDPod(rdSpec.ProtectedPVC.Name)
-	if err != nil {
-		return "", nil, err
-	}
+		// PVC must not be in-use before creating the RD
+		inUse, err := v.isPVCInUseByNonRDPod(rdSpec.ProtectedPVC.Name)
+		if err != nil {
+			return "", nil, requeue, err
+		}
 
-	// It is possible that the PVC becomes in-use at this point (if an app using this PVC is also deployed
-	// on this cluster). That race condition will be ignored. That would be a user error to deploy the
-	// same app in the same namespace and on the destination cluster...
-	if inUse {
-		return "", nil, fmt.Errorf("pvc %v is mounted by others. Checking later", rdSpec.ProtectedPVC.Name)
+		// It is possible that the PVC becomes in-use at this point (if an app using this PVC is also deployed
+		// on this cluster). That race condition will be ignored. That would be a user error to deploy the
+		// same app in the same namespace and on the destination cluster...
+		if inUse {
+			return "", nil, requeue, fmt.Errorf("pvc %v is mounted by others. Checking later", rdSpec.ProtectedPVC.Name)
+		}
+
+		v.log.Info(fmt.Sprintf("Using copyMethod of Direct with App PVC %s", rdSpec.ProtectedPVC.Name))
+		// Using the application PVC for syncing from source to destination and save a snapshot
+		// everytime a sync is successful
+		return volsyncv1alpha1.CopyMethodSnapshot, &rdSpec.ProtectedPVC.Name, !requeue, nil
+	} else if v.IsCopyMethodLocalDirect() {
```

Comment:
I concur. Nonetheless, I refrained from making any changes to the other two options, which will require testing both on my part and by the QE team. As a result, we'll retain the IF/ELSE structure until we conduct the necessary refactoring. I'm uncertain whether we should continue offering the other two options at all. The primary goal of using VolSync with Ramen is to address the limitations of CephFS.

Reply:
I do think we should keep snapshot at a minimum, as it can be really useful for anyone with different storage, or possibly mixed storage.  But, maybe the best argument from the ODF point of view for keeping Snapshot will be if CephFS addresses the snapshot limitations in the future, then a lot of the workaround code could be removed and we could go back to snapshot.